src/retry/mempool.rs

```diff
@@ -0,0 +1,102 @@
+use super::validate::KakarotTransactionValidator;
+use crate::eth_provider::provider::EthDataProvider;
+use reth_transaction_pool::{
+    blobstore::NoopBlobStore, CoinbaseTipOrdering, EthPooledTransaction, Pool, TransactionPool,
+};
+use serde_json::Value;
+use starknet::core::types::Felt;
+use std::{collections::HashSet, fs::File, io::Read, time::Duration};
+use tokio::runtime::Handle;
+
+/// A type alias for the Kakarot Transaction Validator.
+/// Uses the Reth implementation [`TransactionValidationTaskExecutor`].
+pub type Validator<Client> = KakarotTransactionValidator<Client, EthPooledTransaction>;
+
+/// A type alias for the Kakarot Transaction Ordering.
+/// Uses the Reth implementation [`CoinbaseTipOrdering`].
+pub type TransactionOrdering = CoinbaseTipOrdering<EthPooledTransaction>;
+
+/// A type alias for the Kakarot Sequencer Mempool.
+pub type KakarotPool<Client> = Pool<Validator<Client>, TransactionOrdering, NoopBlobStore>;
+
+#[derive(Debug, Default)]
+pub struct AccountManager {
+    accounts: HashSet<Felt>,
+}
```

Comment:
I'm a bit lost at to what this is doing, is this basically the "executor" that fetches the best transactions and sends them?

Reply:
Should the mempool be here then? instead of being with the provider